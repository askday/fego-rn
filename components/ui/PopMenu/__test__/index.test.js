import React from 'react'
import Renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import { View, Text } from 'react-native'
import PopMenu from '../index'

describe('PopMenu Tests', () => {
	it('renders correctly', () => {
		const content = [
			<PopMenu.Item text='添加新朋友' iconName='user'/>,
			<PopMenu.Item iconName='user'><Text>扫一扫</Text></PopMenu.Item>,
			<PopMenu.Item iconName='user'><View><Text>帮助</Text></View></PopMenu.Item>
		];
		const tree = Renderer.create(<PopMenu content={content}
			placement='bottomLeft'>
			点我显示 popover 哦
		</PopMenu>).toJSON();
		
		expect(tree).toMatchSnapshot();
	});
})
