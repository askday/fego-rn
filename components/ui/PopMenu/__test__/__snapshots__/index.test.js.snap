// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopMenu Tests renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={Object {}}
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={Object {}}
    testID={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={Object {}}
    >
      点我显示 popover 哦
    </Text>
  </View>
</View>
`;
